Цель работы: проанализировать выбранный датасет по играм из стима, используя библиотеку pandas и работая с файлом типа Jupyter Notebook. А именно нужно сделать: 
вывести датасет и описать его, 
вывести размерность массива, 
вывести наименование всех колонок, 
написать что они означают в этом датасете, 
вывести все уникальные значения, 
отсортировать по определенным параметрам, 
удалить ненужные столбцы или строчки, 
заменить определенные значения в датасете (например, пустые), 
удалить дубликаты, 
провести анализ с помощью функций info, 
провести анализ с помощью функций describe, 
провести выборку данных по строкам и столбцам с помощью loc, 
сохранить новый датасет, сделать вывод по каждому пункту и описать все.


In [1]:
import pandas as pn
import numpy as np
steam = pn.read_csv('steam_games.csv')

steam

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                            desc_snippet  \
0      Now includes all three premium DLC packs (Unto...   
1      PLAYERUNKNOWN'S BATTLEGROUNDS is a battle roya...   
2      Take command of your own mercenary outfit of '...   
3      The post-soviet country of Chernarus is struck...   
4      EVE Online is a community-driven spaceship MMO...   
...                                                  ...   
40828                                                NaN   
40829                                                NaN   
40830                                                NaN   
40831                                                NaN   
40832  An RPG about a tribe exploring an ancient worl...   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                                NaN   
40829                                                NaN   
40830                                                NaN   
40831                                                NaN   
40832                                                NaN   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                                NaN  Feb 12, 2019   
40829                                                NaN   Feb 5, 2019   
40830                                                NaN  Jul 31, 2018   
40831                                                NaN  Jul 24, 2018   
40832                                                NaN   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUB

Это датасет по играм из стима, он содержит ссылку на игру, её тип, название, краткое описание игры, недавние отзывы, все отзывы, дата релиза, разработчик, издатель, популярные тэги, игровые детали, языки, достижения, жанры, описание игры, наличие взрослого контента, минимальные требования, рекомендованные требования, оригинальная цена, цена со скидкой.

In [3]:
steam.shape


(40833, 20)

Размерность массива

In [4]:
steam.columns


Index(['url', 'types', 'name', 'desc_snippet', 'recent_reviews', 'all_reviews',
       'release_date', 'developer', 'publisher', 'popular_tags',
       'game_details', 'languages', 'achievements', 'genre',
       'game_description', 'mature_content', 'minimum_requirements',
       'recommended_requirements', 'original_price', 'discount_price'],
      dtype='object')

url - ссылка на игру
types - тип (игра, набор, дополнение) 
name - название игры
desc_snippet - краткое описание (небольшой кусочек описания игры)
recent_reviews - недавние отзывы
all_reviews - все отзывы
release_date - дата релиза
developer - разработчик
publisher - издатель
popular_tags - популярные тэги
game_details - игровые детали
languages - доступные языки
achivements - количество достижений
genre - жанры
game_description - описание игры
mature_content - наличие взрослого контента
minimum_requirements - минимальные требования
recommended_requirements - рекомендуемые требования
original_price - оригинальная цена
discount_price - цена по скидке

In [5]:
steam.nunique()

url                         40833
types                           3
name                        40749
desc_snippet                27410
recent_reviews               1299
all_reviews                  9209
release_date                 4107
developer                   17420
publisher                   15290
popular_tags                20852
game_details                 6016
languages                    5496
achievements                  360
genre                        1768
game_description            37496
mature_content               1267
minimum_requirements        14762
recommended_requirements    13836
original_price                681
discount_price               2060
dtype: int64

Уникальные значения во всём датасете

In [6]:
steam['url'].unique()


array(['https://store.steampowered.com/app/379720/DOOM/',
       'https://store.steampowered.com/app/578080/PLAYERUNKNOWNS_BATTLEGROUNDS/',
       'https://store.steampowered.com/app/637090/BATTLETECH/', ...,
       'https://store.steampowered.com/app/906840/Fantasy_Grounds__Quests_of_Doom_4_A_Midnight_Council_of_Quail_5E/',
       'https://store.steampowered.com/app/906635/Mega_Man_X5_Sound_Collection/',
       'https://store.steampowered.com/app/908000/Stories_In_Stone/'],
      dtype=object)

Уникальные значения в столбце url

In [7]:
steam['types'].unique()


array(['app', 'bundle', 'sub', nan], dtype=object)

Уникальные значения для столбца types


In [8]:
steam['name'].unique()


array(['DOOM', "PLAYERUNKNOWN'S BATTLEGROUNDS", 'BATTLETECH', ...,
       'Fantasy Grounds - Quests of Doom 4: A Midnight Council of Quail (5E)',
       'Mega Man X5 Sound Collection', 'Stories In Stone'], dtype=object)

Уникальные значения для столбца name

In [9]:
steam['desc_snippet'].unique()


array(['Now includes all three premium DLC packs (Unto the Evil, Hell Followed, and Bloodfall), maps, modes, and weapons, as well as all feature updates including Arcade Mode, Photo Mode, and the latest Update 6.66, which brings further multiplayer improvements as well as revamps multiplayer progression.',
       "PLAYERUNKNOWN'S BATTLEGROUNDS is a battle royale shooter that pits 100 players against each other in a struggle for survival. Gather supplies and outwit your opponents to become the last person standing.",
       "Take command of your own mercenary outfit of 'Mechs and the MechWarriors that pilot them, struggling to stay afloat as you find yourself drawn into a brutal interstellar civil war.",
       ...,
       'Galactis is a single-player space shooter where good reflexes are in high demand. Revive the glory of the ancient arcade days!',
       'Welcome to Alive , a Third person Cover based shooter. Bring in your friends and join the battle to become the best player, you wi

Уникальные значения для столбца desc_snippet


In [10]:
steam['recent_reviews'].unique()

array(['Very Positive,(554),- 89% of the 554 user reviews in the last 30 days are positive.',
       'Mixed,(6,214),- 49% of the 6,214 user reviews in the last 30 days are positive.',
       'Mixed,(166),- 54% of the 166 user reviews in the last 30 days are positive.',
       ...,
       'Very Negative,(20),- 15% of the 20 user reviews in the last 30 days are positive.',
       'Very Negative,(13),- 7% of the 13 user reviews in the last 30 days are positive.',
       'Mixed,(40),- 50% of the 40 user reviews in the last 30 days are positive.'],
      dtype=object)

Уникальные значения для столбца recent_reviews


In [11]:
steam['all_reviews'].unique()


array(['Very Positive,(42,550),- 92% of the 42,550 user reviews for this game are positive.',
       'Mixed,(836,608),- 49% of the 836,608 user reviews for this game are positive.',
       'Mostly Positive,(7,030),- 71% of the 7,030 user reviews for this game are positive.',
       ...,
       'Mixed,(2,044),- 55% of the 2,044 user reviews for this game are positive.',
       'Very Positive,(106),- 99% of the 106 user reviews for this game are positive.',
       'Very Positive,(469),- 96% of the 469 user reviews for this game are positive.'],
      dtype=object)

Уникальные значения для столбца all_reviews


In [12]:
steam['release_date'].unique()


array(['May 12, 2016', 'Dec 21, 2017', 'Apr 24, 2018', ...,
       'Q3 2019 wishlist', 'EM BREVE', '2019/2020'], dtype=object)

Уникальные значения для столбца release_date


In [13]:
steam['developer'].unique()


array(['id Software', 'PUBG Corporation', 'Harebrained Schemes', ...,
       'Gustavo Contreras', 'OldGHZ', '16 Bit Psych,Kyle B'], dtype=object)

Уникальные значения для столбца developer

In [14]:
steam['publisher'].unique()


array(['Bethesda Softworks,Bethesda Softworks',
       'PUBG Corporation,PUBG Corporation',
       'Paradox Interactive,Paradox Interactive', ...,
       'Gustavo Contreras,Gustavo Contreras', 'OldGHZ,OldGHZ',
       'Self-Publish,Self-Publish'], dtype=object)

Уникальные значения для столбца publisher

In [15]:
steam['popular_tags'].unique()


array(['FPS,Gore,Action,Demons,Shooter,First-Person,Great Soundtrack,Multiplayer,Singleplayer,Fast-Paced,Sci-fi,Horror,Classic,Atmospheric,Difficult,Blood,Remake,Zombies,Co-op,Memes',
       'Survival,Shooter,Multiplayer,Battle Royale,PvP,FPS,Third-Person Shooter,Action,Online Co-Op,Tactical,Co-op,First-Person,Early Access,Strategy,Competitive,Third Person,Team-Based,Difficult,Simulation,Stealth',
       'Mechs,Strategy,Turn-Based,Turn-Based Tactics,Sci-fi,Turn-Based Strategy,Tactical,Singleplayer,Robots,RPG,Action,Multiplayer,Futuristic,Character Customization,Management,Adventure,Space,Story Rich,Great Soundtrack,Difficult',
       ...,
       'Indie,Adventure,Casual,Simulation,Horror,Atmospheric,Puzzle,Replay Value,Singleplayer,Difficult,Point & Click,Sci-fi,Hacking,Experimental,Steampunk,Nonlinear,Multiple Endings,Hidden Object,Cyberpunk,Great Soundtrack',
       'RPG,Strategy,Indie,Software,Turn-Based,Fantasy,Online Co-Op,Party-Based RPG,Adventure',
       'Free to Play,Strategy,A

Уникальные значения для столбца popular_tags

In [16]:
steam['game_details'].unique()


array(['Single-player,Multi-player,Co-op,Steam Achievements,Steam Trading Cards,Partial Controller Support,Steam Cloud',
       'Multi-player,Online Multi-Player,Stats',
       'Single-player,Multi-player,Online Multi-Player,Cross-Platform Multiplayer,Steam Achievements,Steam Trading Cards,Steam Cloud',
       ...,
       'Full controller support,Steam is learning about this software\xa0\r\n\t\t\t\t\t\t\t\t\t',
       'Single-player,Multi-player,Online Multi-Player,Steam Achievements,Steam Cloud,Stats,Steam is learning about this game\xa0\r\n\t\t\t\t\t\t\t\t\t',
       'Single-player,Multi-player,Online Multi-Player,Local Multi-Player,Co-op,Online Co-op,Local Co-op,Shared/Split Screen,Partial Controller Support,Steam Cloud'],
      dtype=object)

Уникальные значения для столбца game_details

In [17]:
steam['languages'].unique()


array(['English,French,Italian,German,Spanish - Spain,Japanese,Polish,Portuguese - Brazil,Russian,Traditional Chinese',
       'English,Korean,Simplified Chinese,French,German,Spanish - Spain,Arabic,Japanese,Polish,Portuguese,Russian,Turkish,Thai,Italian,Portuguese - Brazil,Traditional Chinese,Ukrainian',
       'English,French,German,Russian', ...,
       'English,French,Italian,German,Spanish - Spain,Dutch,Greek,Turkish,Portuguese,Portuguese - Brazil,Russian,Simplified Chinese,Traditional Chinese,Japanese,Korean',
       'English,French,Italian,German,Spanish - Spain,Portuguese - Brazil,Dutch',
       'English,French,German,Spanish - Spain,Simplified Chinese,Korean,Russian,Czech'],
      dtype=object)

Уникальные значения для столбца languages

In [18]:
steam['achievements'].unique()

array([5.400e+01, 3.700e+01, 1.280e+02,       nan, 5.100e+01, 5.500e+01,
       3.400e+01, 4.300e+01, 7.200e+01, 4.100e+01, 5.000e+01, 1.200e+01,
       3.200e+01, 7.500e+01, 6.800e+01, 3.000e+01, 9.600e+01, 1.500e+01,
       8.500e+01, 6.300e+01, 1.400e+01, 4.700e+01, 5.800e+01, 4.600e+01,
       6.100e+01, 7.700e+01, 3.100e+01, 1.470e+02, 7.100e+01, 4.500e+01,
       2.000e+01, 3.800e+01, 9.100e+01, 4.000e+01, 4.800e+01, 6.200e+01,
       1.340e+02, 2.200e+01, 1.900e+01, 5.300e+01, 6.900e+01, 1.600e+01,
       3.500e+01, 1.000e+01, 6.600e+01, 8.000e+01, 4.900e+02, 4.200e+01,
       2.800e+01, 4.400e+01, 9.300e+01, 8.900e+01, 6.700e+01, 1.370e+02,
       4.900e+01, 1.390e+02, 5.900e+01, 5.600e+01, 1.180e+02, 1.800e+01,
       2.400e+01, 7.600e+01, 1.000e+02, 1.100e+01, 3.610e+02, 3.900e+01,
       9.900e+01, 5.200e+01, 4.520e+02, 1.290e+02, 2.900e+01, 1.000e+00,
       1.060e+02, 3.600e+01, 1.440e+02, 7.090e+02, 7.000e+01, 9.700e+01,
       3.300e+01, 1.175e+03, 2.600e+01, 2.620e+02, 

Уникальные значения для столбца achievements

In [19]:
steam['genre'].unique()


array(['Action', 'Action,Adventure,Massively Multiplayer',
       'Action,Adventure,Strategy', ...,
       'Action,Free to Play,Massively Multiplayer,Strategy,Early Access',
       'Action,Free to Play,Indie,Massively Multiplayer,Racing,RPG,Simulation',
       'Casual,Free to Play,Indie,Massively Multiplayer,Simulation,Sports'],
      dtype=object)

Уникальные значения для столбца genre

In [20]:
steam['game_description'].unique()

array([' About This Game Developed by id software, the studio that pioneered the first-person shooter genre and created multiplayer Deathmatch, DOOM returns as a brutally fun and challenging modern-day shooter experience. Relentless demons, impossibly destructive guns, and fast, fluid movement provide the foundation for intense, first-person combat – whether you’re obliterating demon hordes through the depths of Hell in the single-player campaign, or competing against your friends in numerous multiplayer modes. Expand your gameplay experience using DOOM SnapMap game editor to easily create, play, and share your content with the world. STORY: You’ve come here for a reason. The Union Aerospace Corporation’s massive research facility on Mars is overwhelmed by fierce and powerful demons, and only one person stands between their world and ours.  As the lone DOOM Marine, you’ve been activated to do one thing – kill them all. KEY FEATURES: A Relentless Campaign There is no taking cover or sto

Уникальные значения для столбца game_description


In [21]:
steam['mature_content'].unique()


array([nan,
       ' Mature Content Description  The developers describe the content like this:  This Game may contain content not appropriate for all ages, or may not be appropriate for viewing at work: Frequent Violence or Gore, General Mature Content  ',
       ' Mature Content Description  The developers describe the content like this:  WARNING: This game contains strong language, violence and nudity.  ',
       ...,
       ' Mature Content Description  The developers describe the content like this:  Corpses, blood, guts, decapitation  ',
       ' Mature Content Description  The developers describe the content like this:  The game has alcohol and drugs.  ',
       ' Mature Content Description  The developers describe the content like this:  This game is NOT pornographic. It DOES contain a lot of nudity but it is not sexual or explicit enough to be considered porn. X-MiGuPLAY leans on what you could expect from a cable TV subscription channel at 3am and not what one would expect fro

Уникальные значения для столбца mature_content


In [22]:
steam['minimum_requirements'].unique()


array(['Minimum:,OS:,Windows 7/8.1/10 (64-bit versions),Processor:,Intel Core i5-2400/AMD FX-8320 or better,Memory:,8 GB RAM,Graphics:,NVIDIA GTX 670 2GB/AMD Radeon HD 7870 2GB or better,Storage:,55 GB available space,Additional Notes:,Requires Steam activation and broadband internet connection for Multiplayer and SnapMap',
       'Minimum:,Requires a 64-bit processor and operating system,OS:,64-bit Windows 7, Windows 8.1, Windows 10,Processor:,Intel Core i5-4430 / AMD FX-6300,Memory:,8 GB RAM,Graphics:,NVIDIA GeForce GTX 960 2GB / AMD Radeon R7 370 2GB,DirectX:,Version 11,Network:,Broadband Internet connection,Storage:,30 GB available space',
       'Minimum:,Requires a 64-bit processor and operating system,OS:,64-bit Windows 7 or Higher,Processor:,Intel® Core™ i3-2105 or AMD® Phenom™ II X3 720,Memory:,8 GB RAM,Graphics:,Nvidia® GeForce™ GTX 560 Ti or AMD® ATI Radeon™ HD 5870 (1 GB VRAM),DirectX:,Version 11,Network:,Broadband Internet connection,Storage:,35 GB available space,Sound Ca

Уникальные значения для столбца minimum_requirements

In [23]:
steam['recommended_requirements'].unique()


array(['Recommended:,OS:,Windows 7/8.1/10 (64-bit versions),Processor:,Intel Core i7-3770/AMD FX-8350 or better,Memory:,8 GB RAM,Graphics:,NVIDIA GTX 970 4GB/AMD Radeon R9 290 4GB or better,Storage:,55 GB available space,Additional Notes:,Requires Steam activation and broadband internet connection for Multiplayer and SnapMap',
       'Recommended:,Requires a 64-bit processor and operating system,OS:,64-bit Windows 7, Windows 8.1, Windows 10,Processor:,Intel Core i5-6600K / AMD Ryzen 5 1600,Memory:,16 GB RAM,Graphics:,NVIDIA GeForce GTX 1060 3GB / AMD Radeon RX 580 4GB,DirectX:,Version 11,Network:,Broadband Internet connection,Storage:,30 GB available space',
       'Recommended:,Requires a 64-bit processor and operating system,OS:,64-bit Windows 7 or Higher,Processor:,Intel® Core™ i5-4460 or AMD® FX-4300,Memory:,16 GB RAM,Graphics:,Nvidia® GeForce™ GTX 670 or AMD® Radeon™ R9 285 (2 GB VRAM),DirectX:,Version 11,Network:,Broadband Internet connection,Storage:,35 GB available space,Sound 

Уникальные значения для столбца recommended_requirements

In [24]:
steam['original_price'].unique()


array(['$19.99', '$29.99', '$39.99', '$44.99', 'Free', nan, '$59.99',
       '$14.99', '$49.99', 'Free to Play', '1.020', 'Free To Play',
       '$9.99', '$12.99', '$57.91', '$7.99', '$6.99', '$24.99',
       'Demo Middle-earth™: Shadow of War™', 'Play for Free!', '$13.99',
       '$59.98', '$69.99', '$76.95', '$84.97', '$58.89', '$8.99',
       '$75.95', '$34.99', '$151.89', '$4.99', '$84.93', '$86.92',
       '$20.00', '$15.99', '$9.98',
       'Shadow of the Tomb Raider Free Trial', '$5.99', '$11.99',
       '$32.98', '$21.98',
       'FINAL FANTASY XV WINDOWS EDITION PLAYABLE DEMO', '$28.00', 'UNO',
       '$79.99', '$59.95', '$3.99', '$30.00', '$17.99', '$57.98',
       '$84.98', '$229.48', '$269.69', '$29.98', '$31.99', '$16.99',
       '$52.98', '$39.98', '$41.98', '$1.99', '$99.99', '$2.99', '$10.99',
       '$264.82', '$20.99', 'Demo', 'CrossCode Demo', '$88.94', '$149.99',
       '$36.99', '$18.99', '$19.98', '$89.99', '$25.99', '$116.93',
       'EmergeNYC Demo', '$37.99', '

Уникальные значения для столбца original_price

In [25]:
steam['discount_price'].unique()

array(['$14.99', nan, '$35.18', ..., '$18.67', '$53.51', '$164.97'],
      dtype=object)

Уникальные значения для столбца discount_price

In [26]:
steam.sort_values(by='name')

url   types  \
19455  https://store.steampowered.com/app/439260/BUTT...     app   
10601  https://store.steampowered.com/app/308163/Bare...     app   
7059   https://store.steampowered.com/bundle/1541/Big...  bundle   
27890  https://store.steampowered.com/bundle/5056/Eco...  bundle   
28381  https://store.steampowered.com/bundle/6324/Eli...  bundle   
...                                                  ...     ...   
31321       https://store.steampowered.com/app/976190/_/     app   
34989       https://store.steampowered.com/app/409080/_/     app   
34991       https://store.steampowered.com/app/409050/_/     app   
35169   https://store.steampowered.com/digitalgiftcards/     NaN   
39575      https://store.steampowered.com/app/822230/NA/     app   

                                      name  \
19455           "BUTTS: The VR Experience"   
10601  "Barely Attuned Magic Thingy" Staff   
7059                          "Big" Bundle   
27890                    "EcoGames" Bundle   
28381                 "Elimination" Bundle   
...                                    ...   
31321                                  NaN   
34989                                  NaN   
34991                                  NaN   
35169                                  NaN   
39575                                  NaN   

                                            desc_snippet recent_reviews  \
19455  "BUTTS: The VR Experience" is an animated VR s...            NaN   
10601                                                NaN            NaN   
7059                                 "Big" Bundle bundle            NaN   
27890                           "EcoGames" Bundle bundle            NaN   
28381                        "Elimination" Bundle bundle            NaN   
...                                                  ...            ...   
31321                                                NaN            NaN   
34989                                                NaN            NaN   
34991                                                NaN            NaN   
35169                                                NaN            NaN   
39575  With the Dragon Master game, 1 to 8 players ca...            NaN   

                                             all_reviews  release_date  \
19455  Mostly Positive,(61),- 78% of the 61 user revi...   Feb 3, 2016   
10601  1 user reviews,- Need more user reviews to gen...  Jun 19, 2014   
7059                                                 NaN           NaN   
27890                                                NaN           NaN   
28381                                                NaN           NaN   
...                                                  ...           ...   
31321                                                NaN           NaN   
34989                                                NaN           NaN   
34991                                                NaN           NaN   
35169                                                NaN           NaN   
39575                                                NaN       Q1 2019   

                     developer                publisher  \
19455               Tyler Hurd    Tyler Hurd,Tyler Hurd   
10601      Wild Shadow Studios                      NaN   
7059   Laser Guided Games, LLC  Laser Guided Games, LLC   
27890                 EcoGames                 EcoGames   
28381             ReverseGames             ReverseGames   
...                        ...                      ...   
31321                      NaN                      NaN   
34989                      NaN                      NaN   
34991                      NaN                      NaN   
35169                      NaN                      NaN   
39575                      NaN                  N/A,N/A   

                                            popular_tags  \
19455  Psychological Horror,Casual,VR,Nudity,Memes,Fu...   
10601  Action,Free to Play,RPG,Massively Multiplayer,...   
7059

Сортировка по столбцу name

In [27]:
steam.sort_values(by='achievements')

url types  \
15079  https://store.steampowered.com/app/982880/Pixe...   app   
6987   https://store.steampowered.com/app/814900/VRSa...   app   
6798   https://store.steampowered.com/app/884390/Prof...   app   
20747      https://store.steampowered.com/app/1031920/_/   app   
6642   https://store.steampowered.com/app/1078090/Ecc...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
15079                                      Pixel Drawing   
6987                          三国虎将传VR-Sanguo Warriors VR   
6798           Professor Watts Word Search: Pirates Life   
20747                       Симулятор Сидения у Подъезда   
6642                                           Ecchi Sky   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                            desc_snippet recent_reviews  \
15079  Pixel Drawing is simple puzzle where you need ...            NaN   
6987   This is a sub-era realistic VR game. The game ...            NaN   
6798   Hello Professor Watts here and I need your hel...            NaN   
20747                                               Гыгы            NaN   
6642   Welcome to the incredible world of Ecchi Sky. ...            NaN   
...                                                  ...            ...   
40828                                                NaN            NaN   
40829                                                NaN            NaN   
40830                                                NaN            NaN   
40831                                                NaN            NaN   
40832  An RPG about a tribe exploring an ancient worl...            NaN   

                                             all_reviews  release_date  \
15079  2 user reviews,- Need more user reviews to gen...  Nov 21, 2018   
6987   Mixed,(56),- 69% of the 56 user reviews for th...  Apr 28, 2018   
6798   3 user reviews,- Need more user reviews to gen...  Jun 28, 2018   
20747  Mostly Positive,(33),- 75% of the 33 user revi...  Apr 25, 2019   
6642   Mostly Positive,(10),- 70% of the 10 user revi...  May 30, 2019   
...                                                  ...           ...   
40828                                                NaN  Feb 12, 2019   
40829                                                NaN   Feb 5, 2019   
40830                                                NaN  Jul 31, 2018   
40831                                                NaN  Jul 24, 2018   
40832                                                NaN   Aug 8, 2018   

                     developer                            publisher  \
15079                      Sly                    SlyGames,SlyGames   
6987                      GYVR                            GYVR,GYVR   
6798           Ripknot Systems      Ripknot Systems,Ripknot Systems   
20747        Kavkaz Sila Games  Kavkaz Sila Games,Kavkaz Sila Games   
6642               STREAMWORKS              STREAMWORKS,STREAMWORKS   
...                        ...                                  ...   
40828  Ubisoft - San Francisco                                  NaN   
40829  Ubisoft - San Francisco                                  NaN   
40830      SmiteWorks USA, LLC                                  NaN   
40831          CAPCOM CO., LTD      CAP

Сортировка по столбцу achievements

In [28]:
steam['name']

0                                                     DOOM
1                            PLAYERUNKNOWN'S BATTLEGROUNDS
2                                               BATTLETECH
3                                                     DayZ
4                                               EVE Online
                               ...                        
40828    Rocksmith® 2014 Edition – Remastered – Sabaton...
40829    Rocksmith® 2014 Edition – Remastered – Stone T...
40830    Fantasy Grounds - Quests of Doom 4: A Midnight...
40831                         Mega Man X5 Sound Collection
40832                                     Stories In Stone
Name: name, Length: 40833, dtype: object

Вывод только столбца name

In [29]:
steam[steam['developer']=='CAPCOM Co., Ltd.']

url types  \
6      https://store.steampowered.com/app/601150/Devi...   app   
21     https://store.steampowered.com/app/787480/Phoe...   app   
63            https://store.steampowered.com/sub/281610/   sub   
280    https://store.steampowered.com/app/883710/RESI...   app   
378    https://store.steampowered.com/app/582010/MONS...   app   
...                                                  ...   ...   
23263  https://store.steampowered.com/app/381710/Resi...   app   
25939  https://store.steampowered.com/app/762176/Mons...   app   
26041  https://store.steampowered.com/app/891730/Mons...   app   
28673  https://store.steampowered.com/app/941971/Devi...   app   
29782  https://store.steampowered.com/app/533590/Bann...   app   

                                                    name  \
6                                        Devil May Cry 5   
21     Phoenix Wright: Ace Attorney Trilogy / 逆転裁判123...   
63       RESIDENT EVIL 2 / BIOHAZARD RE:2 Deluxe Edition   
280                     RESIDENT EVIL 2 / BIOHAZARD RE:2   
378                                MONSTER HUNTER: WORLD   
...                                                  ...   
23263                     Resident Evil 0 Costume Pack 1   
25939    Monster Hunter: World - Gesture: Spirit Fingers   
26041                 Monster Hunter: World - Deluxe Kit   
28673                  Devil May Cry 5 - 300000 Red Orbs   
29782                     Banned Footage Vol.1 グロテスクVer.   

                                            desc_snippet  \
6      The ultimate Devil Hunter is back in style, in...   
21     Become Phoenix Wright and experience the thril...   
63     RESIDENT EVIL 2 / BIOHAZARD RE:2 Deluxe Editio...   
280    A deadly virus engulfs the residents of Raccoo...   
378    Welcome to a new world! In Monster Hunter: Wor...   
...                                                  ...   
23263                                                NaN   
25939                                                NaN   
26041                                                NaN   
28673                                                NaN   
29782                                                NaN   

                                          recent_reviews  \
6      Very Positive,(408),- 87% of the 408 user revi...   
21     Very Positive,(47),- 95% of the 47 user review...   
63                                                   NaN   
280    Very Positive,(291),- 94% of the 291 user revi...   
378    Very Positive,(924),- 86% of the 924 user revi...   
...                                                  ...   
23263                                                NaN   
25939                                                NaN   
26041                                                NaN   
28673                                                NaN   
29782                                                NaN   

                                             all_reviews  release_date  \
6      Very Positive,(9,645),- 92% of the 9,645 user ...   Mar 7, 2019   
21     Very Positive,(382),- 97% of the 382 user revi...   Apr 9, 2019   
63                                                   NaN           NaN   
280    Overwhelmingly Positive,(14,492),- 95% of the ...  Jan 24, 2019   
378    Mixed,(62,030),- 68% of the 62,030 user review...   Aug 9, 2018   
...                                                  ...           ...   
23263  Mixed,(34),- 64% of the 34 user reviews for th...  Jan 19, 2016   
25939  3 user reviews,- Need more user reviews to gen...   Oct 4, 2018   
26041  Mixed,(235),- 62% of the 235 user reviews for ...   Aug 9, 2018   
28673  3 user reviews,- Need more user reviews to gen...   Mar 7, 2019   
29782  3 user reviews,- Need more user reviews to gen...  Feb 20, 2017   

              developer                          publisher  \
6      CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
21     CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
63     CAPCOM Co., Ltd

Фильтрация с условием, что разработчик CAPCOM Co., Ltd. 

In [3]:
steam.drop(columns=['mature_content','minimum_requirements','recommended_requirements','desc_snippet','popular_tags'],inplace=True)
steam

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                                NaN   
40829                                                NaN   
40830                                                NaN   
40831                                                NaN   
40832                                                NaN   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                                NaN  Feb 12, 2019   
40829                                                NaN   Feb 5, 2019   
40830                                                NaN  Jul 31, 2018   
40831                                                NaN  Jul 24, 2018   
40832                                                NaN   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUBG Corporation   
2          Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3          Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                          CCP                                  CCP,CCP   
...                        ...                                      ...   
40828  Ubisoft - San Francisco                                      NaN   
40829  Ubisoft - San Francisco                                      NaN   
40830      SmiteWorks USA, LLC                                      NaN   
40831          CAPCOM CO., LTD          CAPCOM CO., LTD,CAPCOM CO., LTD   
40832      16 Bit Psych,Kyle B                Self-Publish,Self-Publish   

                            

Были удалены столбцы mature_content,minimum_requirements,'recommended_requirements,desc_snippet,popular_tags

In [4]:
steam = steam.replace({'achievements':np.nan},'No achievements')
steam = steam.replace({'recent_reviews':np.nan},'No reviews')
steam = steam.replace({'all_reviews':np.nan},'No reviews')
steam = steam.replace({'release_date':np.nan},'No release date')
steam = steam.replace({'release_date':np.nan},'No release date')
steam = steam.replace({'game_description':np.nan},'No game description')
steam = steam.replace({'original_price':np.nan},'No price')
steam = steam.replace({'discount_price':np.nan},'No price')
steam = steam.replace({'types':np.nan}, 'No type')
steam = steam.replace({'popular_tags':np.nan},'No tags')
steam

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                         No reviews   
40829                                         No reviews   
40830                                         No reviews   
40831                                         No reviews   
40832                                         No reviews   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                         No reviews  Feb 12, 2019   
40829                                         No reviews   Feb 5, 2019   
40830                                         No reviews  Jul 31, 2018   
40831                                         No reviews  Jul 24, 2018   
40832                                         No reviews   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUBG Corporation   
2          Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3          Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                          CCP                                  CCP,CCP   
...                        ...                                      ...   
40828  Ubisoft - San Francisco                                      NaN   
40829  Ubisoft - San Francisco                                      NaN   
40830      SmiteWorks USA, LLC                                      NaN   
40831          CAPCOM CO., LTD          CAPCOM CO., LTD,CAPCOM CO., LTD   
40832      16 Bit Psych,Kyle B                Self-Publish,Self-Publish   

                            

Были заменены пустые значения NaN на другие, непустые

In [32]:
steam.drop_duplicates()

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                         No reviews   
40829                                         No reviews   
40830                                         No reviews   
40831                                         No reviews   
40832                                         No reviews   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                         No reviews  Feb 12, 2019   
40829                                         No reviews   Feb 5, 2019   
40830                                         No reviews  Jul 31, 2018   
40831                                         No reviews  Jul 24, 2018   
40832                                         No reviews   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUBG Corporation   
2          Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3          Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                          CCP                                  CCP,CCP   
...                        ...                                      ...   
40828  Ubisoft - San Francisco                                      NaN   
40829  Ubisoft - San Francisco                                      NaN   
40830      SmiteWorks USA, LLC                                      NaN   
40831          CAPCOM CO., LTD          CAPCOM CO., LTD,CAPCOM CO., LTD   
40832      16 Bit Psych,Kyle B                Self-Publish,Self-Publish   

                            

Функция удаляет дубликаты (но дубликатов нет)

In [5]:
steam.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40833 entries, 0 to 40832
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   url               40833 non-null  object
 1   types             40833 non-null  object
 2   name              40817 non-null  object
 3   recent_reviews    40833 non-null  object
 4   all_reviews       40833 non-null  object
 5   release_date      40833 non-null  object
 6   developer         40490 non-null  object
 7   publisher         35733 non-null  object
 8   game_details      40313 non-null  object
 9   languages         40797 non-null  object
 10  achievements      40833 non-null  object
 11  genre             40395 non-null  object
 12  game_description  40833 non-null  object
 13  original_price    40833 non-null  object
 14  discount_price    40833 non-null  object
dtypes: object(15)
memory usage: 4.7+ MB


Анализ с помощью функций info
info() выводит информацию в следующем формате:
Количество непустых значений в каждом столбце
Тип данных каждого столбца
Общее количество столбцов
Количество использованной памяти
Если посмотреть на данные, то можно увидеть количество непустых значений, большинство из всего датасета непустые. Тип каждого столбца это object, всего столбцов 15, а оперативной памяти используется почти 5 Мб.
Можно сделать вывод, что датасет содержит достаточно много ненулевых значений.

In [6]:
steam.describe()

url  types  \
count                                             40833  40833   
unique                                            40833      4   
top     https://store.steampowered.com/app/379720/DOOM/    app   
freq                                                  1  38021   

                     name recent_reviews all_reviews     release_date  \
count               40817          40833       40833            40833   
unique              40749           1300        9210             4108   
top     Game + Soundtrack     No reviews  No reviews  No release date   
freq                    4          38127       12363             3179   

                      developer      publisher   game_details languages  \
count                     40490          35733          40313     40797   
unique                    17420          15290           6016      5496   
top     Ubisoft - San Francisco  Degica,Degica  Single-player   English   
freq                       1041            469           3547     19085   

           achievements   genre     game_description original_price  \
count             40833   40395                40833          40833   
unique              361    1768                37497            682   
top     No achievements  Action  No game description       No price   
freq              28639    2386                 2913           5311   

       discount_price  
count           40833  
unique           2061  
top          No price  
freq            26290

Анализ с помощью функций describe
count в нашем датасете показывает количество непустых значений. Почти все значения значения непустые.
unique в нашем датасете показывает значение уникальных значений. Количество уникальных значений гораздо меньше, чем количество всех значений и непустых значений.
top показывает наиболее часто встречающееся значение. Например, в типах, это значение app, а в отзывах это No reviews.
freq показывает, сколько раз встречается наиболее частое значение. 
Можно сделать вывод, что в типах самое популярное это app, так как она встречается 38021, чаще всего у игры нет отзывов, так как No reviews встречается 38127 в недавних отзывах и 12363 во всех отзывах. Самый часто попадающийся разработчик это Ubisoft - San Francisco, так как он встречается 1041 раз. Самый часто попадающийся издатель это Degica, Degica, так как он встречается 469 раз. Самая популярная игровая деталь это одиночная игра (встречается 3547 раз), самый популярный язык, на котором доступна игра - английский (встречается 19085 раз). В большенстве игр датасета нет достижений (встречается 28639 раз), самый популярный жанр - Action (встречается 2386 раз), у большинства игр нет игрового описания (встречается 2913 раз) и у большинства игр цена не указана.


In [35]:
row_data = steam.loc[0]
row_data

url                   https://store.steampowered.com/app/379720/DOOM/
types                                                             app
name                                                             DOOM
recent_reviews      Very Positive,(554),- 89% of the 554 user revi...
all_reviews         Very Positive,(42,550),- 92% of the 42,550 use...
release_date                                             May 12, 2016
developer                                                 id Software
publisher                       Bethesda Softworks,Bethesda Softworks
game_details        Single-player,Multi-player,Co-op,Steam Achieve...
languages           English,French,Italian,German,Spanish - Spain,...
achievements                                                     54.0
genre                                                          Action
game_description     About This Game Developed by id software, the...
original_price                                                 $19.99
discount_price      

Выбор определенной строки по метке индекса, используя loc

In [36]:
steamset = steam.loc[[0,1,2],['name','developer','publisher']]
steamset

name            developer  \
0                           DOOM          id Software   
1  PLAYERUNKNOWN'S BATTLEGROUNDS     PUBG Corporation   
2                     BATTLETECH  Harebrained Schemes   

                                 publisher  
0    Bethesda Softworks,Bethesda Softworks  
1        PUBG Corporation,PUBG Corporation  
2  Paradox Interactive,Paradox Interactive

Выбор определенных строк и столбцов, используя loc

In [37]:
capcom_steam = steam.loc[steam['developer']=='CAPCOM Co., Ltd.']
capcom_steam

url types  \
6      https://store.steampowered.com/app/601150/Devi...   app   
21     https://store.steampowered.com/app/787480/Phoe...   app   
63            https://store.steampowered.com/sub/281610/   sub   
280    https://store.steampowered.com/app/883710/RESI...   app   
378    https://store.steampowered.com/app/582010/MONS...   app   
...                                                  ...   ...   
23263  https://store.steampowered.com/app/381710/Resi...   app   
25939  https://store.steampowered.com/app/762176/Mons...   app   
26041  https://store.steampowered.com/app/891730/Mons...   app   
28673  https://store.steampowered.com/app/941971/Devi...   app   
29782  https://store.steampowered.com/app/533590/Bann...   app   

                                                    name  \
6                                        Devil May Cry 5   
21     Phoenix Wright: Ace Attorney Trilogy / 逆転裁判123...   
63       RESIDENT EVIL 2 / BIOHAZARD RE:2 Deluxe Edition   
280                     RESIDENT EVIL 2 / BIOHAZARD RE:2   
378                                MONSTER HUNTER: WORLD   
...                                                  ...   
23263                     Resident Evil 0 Costume Pack 1   
25939    Monster Hunter: World - Gesture: Spirit Fingers   
26041                 Monster Hunter: World - Deluxe Kit   
28673                  Devil May Cry 5 - 300000 Red Orbs   
29782                     Banned Footage Vol.1 グロテスクVer.   

                                          recent_reviews  \
6      Very Positive,(408),- 87% of the 408 user revi...   
21     Very Positive,(47),- 95% of the 47 user review...   
63                                            No reviews   
280    Very Positive,(291),- 94% of the 291 user revi...   
378    Very Positive,(924),- 86% of the 924 user revi...   
...                                                  ...   
23263                                         No reviews   
25939                                         No reviews   
26041                                         No reviews   
28673                                         No reviews   
29782                                         No reviews   

                                             all_reviews     release_date  \
6      Very Positive,(9,645),- 92% of the 9,645 user ...      Mar 7, 2019   
21     Very Positive,(382),- 97% of the 382 user revi...      Apr 9, 2019   
63                                            No reviews  No release date   
280    Overwhelmingly Positive,(14,492),- 95% of the ...     Jan 24, 2019   
378    Mixed,(62,030),- 68% of the 62,030 user review...      Aug 9, 2018   
...                                                  ...              ...   
23263  Mixed,(34),- 64% of the 34 user reviews for th...     Jan 19, 2016   
25939  3 user reviews,- Need more user reviews to gen...      Oct 4, 2018   
26041  Mixed,(235),- 62% of the 235 user reviews for ...      Aug 9, 2018   
28673  3 user reviews,- Need more user reviews to gen...      Mar 7, 2019   
29782  3 user reviews,- Need more user reviews to gen...     Feb 20, 2017   

              developer                          publisher  \
6      CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
21     CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
63     CAPCOM Co., Ltd.                   CAPCOM Co., Ltd.   
280    CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
378    CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
...                 ...                                ...   
23263  CAPCOM Co., Ltd.                                NaN   
25939  CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
26041  CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
28673  CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   
29782  CAPCOM Co., Ltd.  CAPCOM Co., Ltd.,CAPCOM Co., Ltd.   

                                            game_details  \
6      Single-player,Online Multi-Player,Online Co-op...   
21     Single-player,Steam A

Выбор строк с определенным условием, используя loc

In [38]:
value = steam.loc[6,'developer']
value

'CAPCOM Co., Ltd.'

Выбор конкретной ячейки данных, используя loc

In [39]:
new_value = steam.loc[6,'developer'] = 'CAPCOM THE BEST GAME EVER'
steam

url types  \
0        https://store.steampowered.com/app/379720/DOOM/   app   
1      https://store.steampowered.com/app/578080/PLAY...   app   
2      https://store.steampowered.com/app/637090/BATT...   app   
3        https://store.steampowered.com/app/221100/DayZ/   app   
4      https://store.steampowered.com/app/8500/EVE_On...   app   
...                                                  ...   ...   
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
0                                                   DOOM   
1                          PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                             BATTLETECH   
3                                                   DayZ   
4                                             EVE Online   
...                                                  ...   
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                          recent_reviews  \
0      Very Positive,(554),- 89% of the 554 user revi...   
1      Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2      Mixed,(166),- 54% of the 166 user reviews in t...   
3      Mixed,(932),- 57% of the 932 user reviews in t...   
4      Mixed,(287),- 54% of the 287 user reviews in t...   
...                                                  ...   
40828                                         No reviews   
40829                                         No reviews   
40830                                         No reviews   
40831                                         No reviews   
40832                                         No reviews   

                                             all_reviews  release_date  \
0      Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1      Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2      Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3      Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4      Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   
...                                                  ...           ...   
40828                                         No reviews  Feb 12, 2019   
40829                                         No reviews   Feb 5, 2019   
40830                                         No reviews  Jul 31, 2018   
40831                                         No reviews  Jul 24, 2018   
40832                                         No reviews   Aug 8, 2018   

                     developer                                publisher  \
0                  id Software    Bethesda Softworks,Bethesda Softworks   
1             PUBG Corporation        PUBG Corporation,PUBG Corporation   
2          Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3          Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                          CCP                                  CCP,CCP   
...                        ...                                      ...   
40828  Ubisoft - San Francisco                                      NaN   
40829  Ubisoft - San Francisco                                      NaN   
40830      SmiteWorks USA, LLC                                      NaN   
40831          CAPCOM CO., LTD          CAPCOM CO., LTD,CAPCOM CO., LTD   
40832      16 Bit Psych,Kyle B                Self-Publish,Self-Publish   

                            

Обновление данных с использованием loc

In [40]:
steam.to_csv('new_steam.csv',index = False)

Вывод: в ходе работы был проанализирован датасет по стиму, была использована библиотека pandas. Были выполнены следующий цели: 
вывести датасет и описать его, 
вывести размерность массива, 
вывести наименование всех колонок, 
написать что они означают в этом датасете, 
вывести все уникальные значения, 
отсортировать по определенным параметрам, 
удалить ненужные столбцы или строчки, 
заменить определенные значения в датасете (например, пустые), 
удалить дубликаты, 
провести анализ с помощью функций info, 
провести анализ с помощью функций describe, 
провести выборку данных по строкам и столбцам с помощью loc, 
сохранить новый датасет, сделать вывод по каждому пункту и описать все.